### PR TITLE
GridView added

### PR DESCRIPTION
**Pull request by ldanilina**
GridView added in two modes: 
- vertically scrollable (number of columns in portrait / landscape mode is configurable), default
- horizontally scrollable (number of rows in portrait / landscape mode is configurable)

### CODE REVIEW

**Comment by CasperMok** [comment]
The tapping behavior on a small title doesn't seem to be correct. We need a way of tapping a small title and transforming it into the corresponding stream view.

**Comment by ldanilina** [comment]
> The tapping behavior on a small title doesn't seem to be correct. We need a way of tapping a small title and transforming it into the corresponding stream view. Let's discuss this.

Got it, fixed, thanks

**Comment by CasperMok** [comment]
The video is transformed into a single view shall be the one a user is tapping on in the grid view small title. It's not like this now. Happy to discuss a better and proper req on this.

File: Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift [hunk 1]
```diff
@@ -327,6 +354,32 @@ final class StreamViewModel: ObservableObject {
             )
 
             displayMode = .list(listViewModel)
+        case .grid:
+            let secondaryVideoSources = sortedSources.filter { $0.id != selectedVideoSource.id }
```
Comment (CasperMok) [inline Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift:358]: There are a lot of similarities to lines 162 to 186 above. Is it possible to group them into a single method?

File: Sources/DolbyIORTSUIKit/Private/Views/GridView/GridView.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                        onVideoSelection(source)
+                    }
+                    .frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .center)
+                }
+            }
+            .frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .center)
+        }
+    }
+
+    private func gridHorizontal(availableHeight: CGFloat, rowsCount: Int) -> some View {
```
Comment (CasperMok) [inline Sources/DolbyIORTSUIKit/Private/Views/GridView/GridView.swift:88]: Minor style for matching gridVertical() method:
`gridHorizontal(_ availableHeight: CGFloat, _ rowsCount: Int)`

File: Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift (hunk 1 above)
Comment (ldanilina) [reply Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift:362]: refactored, thanks

**Comment by ldanilina** [comment]
> The video is transformed into a single view shall be the one a user is tapping on in the grid view small title. It's not like this now. Happy to discuss a better and proper req on this.

Could you please take a look again? Is it as expected now? Thanks!

**Comment by CasperMok** [comment]
The selected small title now will be prompted to be the 1st title in the grid view. I think it may not be necessary. Anyway, we may further discuss and improve this in the future.

**Review by CasperMok — APPROVED** [review]

**Pull request merged**